nautobot/dcim/tests/test_views.py

```diff
@@ -1344,14 +1345,71 @@ def test_device_poweroutlets(self):
     @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
     def test_device_interfaces(self):
         device = Device.objects.first()
+        self.add_permissions("ipam.add_ipaddress", "dcim.change_interface")
 
         intf_status = Status.objects.get_for_model(Interface).first()
-        Interface.objects.create(device=device, name="Interface 1", status=intf_status)
-        Interface.objects.create(device=device, name="Interface 2", status=intf_status)
-        Interface.objects.create(device=device, name="Interface 3", status=intf_status)
+        interfaces = (
+            Interface.objects.create(device=device, name="Interface 1", status=intf_status),
+            Interface.objects.create(device=device, name="Interface 2", status=intf_status),
+            Interface.objects.create(device=device, name="Interface 3", status=intf_status),
+        )
 
         url = reverse("dcim:device_interfaces", kwargs={"pk": device.pk})
-        self.assertHttpStatus(self.client.get(url), 200)
+        response = self.client.get(url)
+        self.assertHttpStatus(response, 200)
+        response_body = response.content.decode(response.charset)
+        # Count the number of occurrences of "Add IP address" in the response_body
+        count = response_body.count("Add IP address")
+        # Assert that "Add IP address" appears three times for each of the interfaces
+        self.assertEqual(count, 3)
+
+        with self.subTest("Assert Create and Assign IPAddress"):
+            # Assest assigning ipadress to interface
+            namespace = Namespace.objects.create(name="ipam_test_views_ip_address_test")
+            prefix_status = Status.objects.get_for_model(Prefix).first()
+            Prefix.objects.get_or_create(
+                prefix="192.0.2.0/24",
+                defaults={"namespace": namespace, "status": prefix_status, "type": "network"},
+            )
+            form_data = {
+                "namespace": namespace.pk,
+                "address": IPNetwork("192.0.2.99/24"),
```

Comment:
Does `"address": "192.0.2.99/24"` not work? I'd expect the post data to just be a string.